ADR/0022-secret-mgmt-for-user-workloads.md

```diff
@@ -10,6 +10,9 @@ Accepted
 
 * When user workloads are deployed to environments, the system should be able to provide a way to inject values that are specific to the environment. Currently, this is done through environment variables that are managed as overlays on the GitOps repository for the application. However, this method does not provide a good way to manage `Secret`. This ADR addresses the secret management of user workloads for different environments.
 * As a StoneSoup component, I expect to have the ability to securely upload a secret, associate it with the component, and propagate it to the target environment.
+* It should be possible to distinguish in time the process of uploading confidential information and deploying it in the destination environment. For example, image-controller creates an image pull secret after `Component` is created, and deploys it after `Environment` is made.
```

Comment:
I'd like to have example in this item clarified.
I completely agree that credentials upload and their further provision could be separated in time and should be supported. However, the task of Image Controller is to provision image repository and upload its credentials to SPI. So, in general, selecting target environments should not be done by Image Controller.